src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs

```diff
@@ -1222,49 +1279,89 @@ public static AkkaConfigurationBuilder WithSingleton<TKey>(
+        ///     </para>
+        ///     If <paramref name="createProxyToo"/> is set to <c>true</c> then this method will also create a
+        ///     <see cref="ClusterSingletonProxy"/> that will be added to the <see cref="ActorRegistry"/> using the key
+        ///     <see cref="TKey"/>. Otherwise, this method will register nothing with the <see cref="ActorRegistry"/>.
+        ///     <list type="bullet">
+        ///         <item>
+        ///             The complete singleton manager actor path name will be
+        ///             <c>akka://MyActorSystem/user/singletonManagerName/singletonName</c>
+        ///         </item>
+        ///         <item>
+        ///             The complete singleton proxy actor path name, if created, will be
+        ///             <c>akka://MyActorSystem/user/singletonName-proxy</c>
+        ///         </item>
+        ///     </list>
+        /// </summary>
+        /// <param name="builder">
+        ///     The builder instance being configured.
+        /// </param>
+        /// <param name="singletonManagerName">
+        ///     The name of the <see cref="ClusterSingletonManager"/> created by this method. See summary above.
+        /// </param>
+        /// <param name="singletonName">
+        ///     The name of this singleton instance and optionally, part of the <see cref="ClusterSingletonProxy"/>
+        ///     name created by this method. See summary above.
+        /// </param>
+        /// <param name="propsFactory">
+        ///     A function that accepts the <see cref="ActorSystem"/>, <see cref="ActorRegistry"/>, and <see cref="IDependencyResolver"/>
+        ///     and returns the <see cref="Props"/> for the actor
+        /// </param>
+        /// <param name="options">
+        ///     Optional. The set of options for configuring both the <see cref="ClusterSingletonManager"/> and
+        ///     optionally, the <see cref="ClusterSingletonProxy"/>.
+        /// </param>
+        /// <param name="createProxyToo">
+        ///     When set to <c>true></c>, creates a <see cref="ClusterSingletonProxy"/> that automatically points to
+        ///     the <see cref="ClusterSingletonManager"/> created by this method.
+        /// </param>
+        /// <typeparam name="TKey">
+        ///     The key type to use for the <see cref="ActorRegistry"/> when <paramref name="createProxyToo"/> is set to <c>true</c>.
+        /// </typeparam>
+        /// <returns>
+        ///     The same <see cref="AkkaConfigurationBuilder"/> instance originally passed in.
+        /// </returns>
+        public static AkkaConfigurationBuilder WithSingleton<TKey>(
+            this AkkaConfigurationBuilder builder,
+            string singletonManagerName,
+            string singletonName,
+            Func<ActorSystem, IActorRegistry, IDependencyResolver, Props> propsFactory,
+            ClusterSingletonOptions? options = null,
+            bool createProxyToo = true)
```

Comment:
New API that allows users to specify the singleton manager actor name.

The resulting singleton path will be `akka://MyActorSystem/user/singletonManagerName/singletonName`

Old APIs are untouched, they will still generate the same singleton actor path `akka://MyActorSystem/user/singletonName/singletonName`